File: 백준/Silver/10773. 제로/제로.py
import sys
from collections import deque
inputCount = int(sys.stdin.readline())
q = deque()
for i in range(inputCount) :
    number = int(sys.stdin.readline())
    if number == 0 :
        q.pop()
    else :
        q.append(number)
print(sum(list(q)))

File: 백준/Silver/10773. 제로/제로.cs
internal class Program
{
    static void Main(string[] args)
    {
        new BaekJoon_10773().Start();
    }
}
public class BaekJoon_10773
{
    public void Start()
    {
        int K = int.Parse(Console.ReadLine());
        Stack<int> s = new Stack<int>();
        for (int i = 0; i < K; i++)
        {
            int input = int.Parse(Console.ReadLine());
            if(input == 0 && s.Count > 0)
            {
                s.Pop();
            }
            else
            {
                s.Push(input);
            }
            
        }
        Console.WriteLine(s.Sum());
    }
    
}